spec/integration/api/transform_spec.rb

```diff
@@ -0,0 +1,247 @@
+require 'spec_helper'
+require 'securerandom'
+
+module Vault
+  describe Transform, ent_vault: ">= 1.4" do
+    subject { vault_test_client }
+
+    before(:context) do
+      vault_test_client.sys.mount(
+        "transform", "transform", "transform"
+      )
+      vault_test_client.transform.create_role("foo_role", transformations: ["foo_trans"])
+      vault_test_client.transform.create_transformation("foo_trans", type: "fpe", template: "builtin/creditcardnumber", allowed_roles: ["foo_role"])
+    end
+
+    # Lists of possible input options for a given transform type
+    base_names = Proc.new { SecureRandom.alphanumeric(5) }
+    trans_types = ['fpe', 'masking']
+    trans_tweak_sources = ['supplied', 'generated', 'internal']
+    trans_templates = ['builtin/creditcardnumber', 'builtin/socialsecuritynumber']
```

Comment:
There are also builtin alphabets, not sure if you'd want a test for them as well or not: https://www.vaultproject.io/docs/secrets/transform#alphabets

Reply:
I mainly was checking to see if new ones could be created with different sets of options.